hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/controller/ContractController.java

```diff
@@ -61,7 +105,23 @@ private Mono<GenericErrorResponse> unsupportedOpResponse(UnsupportedOperationExc
     @ExceptionHandler
     @ResponseStatus(BAD_REQUEST)
     private Mono<GenericErrorResponse> validationError(WebExchangeBindException e) {
-        return errorResponse(parseValidationError(e));
+        final var errorMessage = parseValidationError(e);
+        log.warn("Validation error: {}", errorMessage);
+        return errorResponse(errorMessage);
+    }
+
+    @ExceptionHandler
+    @ResponseStatus(BAD_REQUEST)
+    private Mono<GenericErrorResponse> invalidTxnError(InvalidTransactionException e) {
+        log.warn("Transaction error: {}", e.getMessage());
+        return errorResponse(e.getMessage());
+    }
+
+    @ExceptionHandler()
+    @ResponseStatus(INTERNAL_SERVER_ERROR)
+    private Mono<GenericErrorResponse> genericError(Exception e) {
+        log.warn("Generic error: {}", e.getMessage());
```

Comment:
We should log at error for internal errors and print the stacktrace.
```suggestion
        log.error("Generic error: ", e);
```